router/link/link_registry.go

```diff
@@ -165,9 +165,9 @@ func (self *linkRegistryImpl) LinkClosed(link xlink.Xlink) {
 	defer self.Unlock()
 	if val := self.linkMap[link.Key()]; val == link {
 		delete(self.linkMap, link.Key())
+		self.updateLinkStateClosed(link) // only update link state to closed if this was the current link
```

Comment:
should this actually happen before the map delete? probably isn't important but figured i'd ask to make you consider it. "it doesn't matter" is fine for me if that's the case

Reply:
It doesn't matter as map access is protected by the mutex above. If it did matter, I'd want the link to be cleared first before kicking the redial process.

Reply:
👍 couldn't see the mutex without expanding the lines :) thx